### PR TITLE
chore: remove sdkv2 from cloudformation-diff

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
### Issue # (if applicable)

Closes #<issue number here>.

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts [hunk 1]
```diff
@@ -1104,7 +1104,7 @@ describe('changeset', () => {
             ResourceType: 'AWS::S3::Bucket',
             Replacement: 'True',
             Details: [{
-              Evaluation: 'Direct',
+              Evaluation: 'Static',
```
Comment (piradeepk) [inline packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts:1107]: Curious, do you mind explaining why we needed to change the Evaluation type from Direct to Static?

File: packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 // The SDK should not make network calls here
-import type { CloudFormation } from 'aws-sdk';
+import type { DescribeChangeSetOutput as DescribeChangeSet } from '@aws-sdk/client-cloudformation';
 import * as impl from './diff';
 import * as types from './diff/types';
 import { deepEqual, diffKeyedEntities, unionOf } from './diff/util';
 
 export * from './diff/types';
 
+export type DescribeChangeSetOutput = DescribeChangeSet;
```
Comment (comcalvi) [inline packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts:10]: why both this and the import? The change on line 3 lets us reference this type as `DescribeChangeSet`, but without the `as`, you could achieve the same result we get from both lines 3 and 10.

File: yarn.lock [hunk 3]
```diff
@@ -1848,6 +2113,19 @@
     "@smithy/types" "^2.5.0"
     tslib "^2.5.0"
 
+"@aws-sdk/credential-provider-sso@3.540.0":
+  version "3.540.0"
```
Comment (comcalvi) [inline yarn.lock:2117]: I guess these are fine? ideally the changes to `yarn.lock` would reflect only the diff in `package.json`, but node dependency tooling is funny

**Review by comcalvi — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts (hunk 1 above)
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/cloudformation-diff/test/diff-template.test.ts:1107]: This was not a valid value before but in v2 it wasn't strongly typed enough for it to matter in our test case.

File: packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts (hunk 2 above)
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts:10]: This type is used extensively throughout aws-cdk (the cli, not the entire package). This re-export is a workaround In order to not cause a conflict of types in that package. You could argue that we should just migrate that whole package, but that's a much more complex task that we have not yet prioritized.

**Review by comcalvi — APPROVED** [review]

**Review by piradeepk — APPROVED** [review]
Lgtm! 🚀

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Pull request merged**